dask/dataframe/core.py

```diff
@@ -4269,24 +4362,15 @@ class DataFrame(_Frame):
     _token_prefix = "dataframe-"
     _accessors: ClassVar[set[str]] = set()
 
-    def __init__(self, dsk, name, meta, divisions):
-        super().__init__(dsk, name, meta, divisions)
-        if self.dask.layers[name].collection_annotations is None:
-            self.dask.layers[name].collection_annotations = {
-                "npartitions": self.npartitions,
-                "columns": [col for col in self.columns],
-                "type": typename(type(self)),
-                "dataframe_type": typename(type(self._meta)),
-                "series_dtypes": {
-                    col: self._meta[col].dtype
-                    if hasattr(self._meta[col], "dtype")
-                    else None
-                    for col in self._meta.columns
-                },
-            }
-        else:
-            self.dask.layers[name].collection_annotations.update(
-                {
+    def __init__(self, dsk=None, name=None, meta=None, divisions=None, operation=None):
+        super().__init__(
+            dsk=dsk, name=name, meta=meta, divisions=divisions, operation=operation
+        )
+
+        # TODO: Get "collection_annotations" info from operation
```

Comment:
I suppose we should start planning early for the `CollectionOperation` version of annotations/collection_annotations

Reply:
Sigh...